### PR TITLE
Add SnapLink tests and additional exception context for SnapLink failures

### PR DESCRIPTION
**Pull request by dekobon**
This PR adds some minor improvements to tests and exception messages for SnapLinks. These changes were spawned out of the investigation of a bug reporded with the Manta Hadoop Filesystem driver.

### CODE REVIEW

**Comment by cburroughs** [comment]
If #252 has turned out to be a bug elsewhere, can this be closed?

**Comment by dekobon** [comment]
#252 turned out to not be a bug, but I do think we should take these changes. @cburroughs thoughts?

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientIT.java [hunk 1]
```diff
@@ -482,6 +512,7 @@ public final void testList() throws IOException {
     }
 
     @Test(expectedExceptions = MantaObjectException.class)
+    @SuppressWarnings("ReturnValueIgnored")
```
Comment (cburroughs) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaClientIT.java]: Is this related?

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientIT.java [hunk 2]
```diff
@@ -494,6 +525,7 @@ public final void testListNotADir() throws IOException {
     }
 
     @Test(expectedExceptions = MantaClientHttpResponseException.class)
+    @SuppressWarnings("ReturnValueIgnored")
```
Comment (cburroughs) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaClientIT.java]: Is this related?

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientIT.java (hunk 1 above)
Comment (dekobon) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaClientIT.java]: I had to add these annotations to get the library to compile on my platform.

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientIT.java (hunk 2 above)
Comment (dekobon) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaClientIT.java]: I had to add these annotations to get the library to compile on my platform.

**Comment by dekobon** [comment]
> I think for most of the existing funky character tests I just did ~!@#$%^&*()_+ all at once. If you prefer to break it down into smaller tests I understand the tradeoff, but I'm not sure I see a reason to test an arbitrary smaller set of troublesome characters for snaplinks than for other api calls.

The tests that matter are the tests that verify that the encoding works for the location header because it uses a different code path on the server and we want to verify that deals with encoded characters correctly.

**Comment by cburroughs** [comment]
I understand the need the tests for `+=@`. My question is why not also have tests for `~!#$%^&*()` or the troublesome space if it is a different code path.

**Comment by dekobon** [comment]
> My question is why not also have tests for ~!#$%^&*() or the troublesome space if it is a different code path.

Good point. I'll update the PR to reflect that.
